### DISCUSSION TITLE
Kernel produces "...was never added to this object's InputParameters"

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#25753] [Q&A Modules: General] [answered]
Hello,

I am working on a Joule heating source kernel that allows coupling electrical conductivity dynamically and when I run the kernel in a simple electrothermal simulation I get the following error:

*** ERROR ***
Joule_self_heating: The coupled variable "coupled_sigma" was never added to this object's InputParameters, please double-check your spelling

This is the InputParameters section of my kernel.

Any suggestions thx!

John Sevic

InputParameters
JouleHeatingCoupled::validParams()
{
  InputParameters params = Kernel::validParams();

  params.addRequiredCoupledVar(
    "coupled_V", "coupled_sigma");

  return params;
}

JouleHeatingCoupled::JouleHeatingCoupled(const InputParameters & parameters)
  : Kernel(parameters),
    _grad_V(coupledGradient("coupled_V")),
    _D(coupledValue("coupled_sigma"))
{
}

**Comment by GiudGiud** [comment, marked as answer]
this line isnt right

params.addRequiredCoupledVar(
"coupled_V", "coupled_sigma");

this only adds V, with a description of "coupled_sigma"

you need 2 lines

```
params.addRequiredCoupledVar("coupled_V", "The V variable");
params.addRequiredCoupledVar("coupled_sigma", "The sigma variable");
```

**Comment by jsevic** [comment]
Ah great that fixed it,  thx!
